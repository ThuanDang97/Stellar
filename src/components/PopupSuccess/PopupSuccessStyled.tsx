import styled, { keyframes } from 'styled-components'

// Props type
import { TextAnimatedProps } from '@self-types/components/TextAnimated.props'

// Utils
import { pxToRem } from '@utils/theme'

// Components
import { ViewStyled } from '@components/styled-components'

const fadeOut = keyframes`
    from {
      opacity: 1;
    }
    to {
      opacity: 0;
      transform: translateX(${pxToRem(-180)});
    }
`
export const AnimatedPopupSuccess = styled(ViewStyled)<TextAnimatedProps>`
  animation: ${fadeOut} 0.35s linear both 4.65s;
`
