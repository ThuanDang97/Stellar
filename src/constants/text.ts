export const TEXT_FORGOT_PASSWORD =
  'No worries! Enter your email and we will send you a link to reset.'

export const BANNER_TITLE = {
  NO_LINK_BILL: 'You’re Almost Done',
  LINKED_BILL: 'Woohoo! You did it.',
}

export const BANNER_DESCRIPTION = {
  NO_LINK_BILL:
    'Let’s set up your first bill and Stellar will make sure they get paid on time!',
  LINKED_BILL:
    'You got rid of the gravity holding down your credit score. Keep adding bills to boost your score.',
}

export const SUBSCRIPTION_FLEX_DESCRIPTION =
  'Bill credit by paying all your bills on your own flexible schedule.'
export const SUBSCRIPTION_LAUNCH_DESCRIPTION =
  'Autopay all your bills in one place and build your credit the easy way.'

export const CONNECT_BANK_DESCRIPTION_SUCCESS =
  'Your Account has been added. Would you like to link another account?'

export const CONNECT_BANK_DESCRIPTION =
  'Start by linking your checking account to allow us to automatically pay your bills.'

export const SUCCESS = 'Success!'

export const TITLE_CONNECT_BANK = `Let's connect your bank account.`

export const IDENTITY_DESCRIPTION =
  'Please use your legal name. Your date of birth will help us confirm your identity.'

export const DELETE_CONFIRM = {
  DELETE_GOAL: 'Are you sure you want to delete this item?',
  DELETE_LINK_BILL: 'Are you sure you want to delete this linked bill?',
  DELETE_BILL: 'Are you sure you want to delete this bill?',
}

export const DELETE_SUCCESS = {
  GOAL: 'Item has been deleted.',
  LINKED_BILL: 'The linked bill has been deleted.',
}

export const EDIT_SUCCESS = 'Your changes have been successfully saved.'

export const TITLE_CONNECT_ACCOUNT = 'Connect Account'

export const ADD_FIRST_BILL_TITLE = `Congrats, you're about to set up your first bill with Stellar!`
export const ADD_FIRST_BILL_DESC =
  'Once you link the bill, we will pay it on time and draft the funds from your bank account on the selected date.'

export const TITLE_CONFIRM_DETAIL = 'Confirm Details'

export const TITLE_EDIT_LINKED_BILL = 'Edit Linked Bill'

export const SNACKBAR_SUCCESS = {
  ADD_GOAL: 'Your new goals were successfully added.',
  ADD_BILL: 'Your bill was successfully added.',
  EDIT_BILL: 'Your changes have been successfully saved.',
  ADD_LINK_BILL: 'Your bill was successfully linked.',
  DELETE_GOAL: 'Your goal was successfully deleted.',
  DELETE_BILL: 'Your bill was successfully deleted.',
  DELETE_LINK_BILL: 'Your linked bill was successfully deleted.',
}

export const PAY_PERIOD = {
  PAY_MONTHLY: 'Reported Monthly',
  PAY_THREE_MONTHS: 'Reported 3 months',
  PAY_SIX_MONTHS: 'Reported 6 months',
  PAY_ANNUALLY: 'Reported Annually',
}
