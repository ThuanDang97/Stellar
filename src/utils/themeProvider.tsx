import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import { theme } from '@themes/index'

export const renderWithTheme = (component: React.ReactNode) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
}
