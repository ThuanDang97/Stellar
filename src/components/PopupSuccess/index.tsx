import { memo, useEffect } from 'react'
import Image from 'next/image'
import { useTheme } from 'styled-components'

// Props type
import { PopupSuccessProps } from '@self-types/components/PopupSuccess.props'

// Components
import { TextStyled, ViewStyled } from '@components/styled-components'
import { AnimatedPopupSuccess } from '@components/PopupSuccess/PopupSuccessStyled'
import Title from '@components/Title'
import Button from '@components/Button'

const PopupSuccess = ({
  isOpen,
  textContent,
  onClose,
  ...props
}: PopupSuccessProps) => {
  const theme = useTheme()

  useEffect(() => {
    const timerId = setTimeout(() => {
      onClose()
    }, 5000)

    return () => clearTimeout(timerId)
  }, [isOpen, onClose])

  return (
    <>
      {isOpen && (
        <AnimatedPopupSuccess
          position="fixed"
          bottom={props.bottom || theme.metrics.dimensions.md}
          left={props.left}
          display="flex"
          justifyContent="space-between"
          bgColor={theme.colors.firefly}
          borderRadius={theme.metrics.borderRadius.default}
          pLeft={theme.metrics.dimensions.base}
          maxWidth={400}
          data-testid="popup-success"
          right={theme.metrics.dimensions.md}
        >
          <ViewStyled
            display="flex"
            gap={theme.metrics.dimensions.base}
            justifyContent="space-between"
          >
            <Image
              width={theme.metrics.dimensions.lg}
              height={theme.metrics.dimensions.lg}
              src="/icons/check2-circle-light.svg"
            />
            <ViewStyled
              display="flex"
              flexDirection="column"
              justifyContent="flex-start"
            >
              <Title
                headingLevel="h3"
                textAlign="start"
                width={theme.metrics.width.full}
                fontSize={theme.typography.fontSize.lg}
                lineHeight={theme.typography.fontSize.lg}
                letterSpacing={theme.typography.letterSpacing.xxl}
                color={theme.colors.white}
                title="Success"
                fontFamily="AdobeCleanBold"
                pBottom={theme.metrics.dimensions.xs}
              />
              <TextStyled
                as="p"
                textAlign="start"
                width={theme.metrics.width.full}
                fontSize={theme.typography.fontSize.xss}
                lineHeight={theme.typography.fontSize.sm}
                color={theme.colors.silverLight}
                fontWeight={theme.typography.fontWeight.small}
                letterSpacing={theme.typography.letterSpacing.xxl}
              >
                {textContent}
              </TextStyled>
            </ViewStyled>
          </ViewStyled>
          <Button
            variant="dark"
            mBottom={theme.metrics.dimensions.lg}
            onClick={onClose}
            width={theme.metrics.dimensions.base}
            height={theme.metrics.dimensions.base}
            imgUrl="/icons/x-lg.svg"
          />
        </AnimatedPopupSuccess>
      )}
    </>
  )
}

export default memo(PopupSuccess)
