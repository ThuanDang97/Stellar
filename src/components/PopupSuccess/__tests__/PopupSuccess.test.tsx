import { fireEvent, screen } from '@testing-library/react'
import { act } from 'react-dom/test-utils'

// Utils
import { renderWithTheme } from '@utils/themeProvider'

// Constants
import { EDIT_SUCCESS } from '@constants/text'

// Components
import PopupSuccess from '@components/PopupSuccess'

describe('PopupSuccess Component', () => {
  const mockCloseFunct = jest.fn()

  const props = {
    isOpen: true,
    textContent: EDIT_SUCCESS,
    onClose: mockCloseFunct,
  }

  test('renders PopupSuccess Confirm unchanged', () => {
    const { asFragment } = renderWithTheme(<PopupSuccess {...props} />)

    expect(asFragment()).toMatchSnapshot()
  })

  test('runs close function after setTimeout', () => {
    jest.useFakeTimers()

    renderWithTheme(<PopupSuccess {...props} />)
    act(() => {
      jest.runAllTimers()
    })

    expect(mockCloseFunct).toHaveBeenCalled()
  })

  test('runs close function on click icon', () => {
    renderWithTheme(<PopupSuccess {...props} />)

    fireEvent.click(screen.getByTestId('button'))

    expect(mockCloseFunct).toHaveBeenCalled()
  })

  test('do not render PopupSuccess when it is not open', () => {
    renderWithTheme(<PopupSuccess {...props} isOpen={false} />)

    expect(screen.queryByTestId('popup-success')).not.toBeInTheDocument()
  })
})
