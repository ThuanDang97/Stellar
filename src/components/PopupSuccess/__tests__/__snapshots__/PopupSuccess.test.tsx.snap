// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopupSuccess Component renders PopupSuccess Confirm unchanged 1`] = `
<DocumentFragment>
  <div
    class="ViewStyled-sc-rqe5sa-0 PopupSuccessStyled__AnimatedPopupSuccess-sc-1umv97q-0 cghjty bnNRjk"
    data-testid="popup-success"
    display="flex"
  >
    <div
      class="ViewStyled-sc-rqe5sa-0 kgTlIu"
      display="flex"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2730%27%20height=%2730%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div
        class="ViewStyled-sc-rqe5sa-0 eaFPBc"
        display="flex"
      >
        <h3
          class="TextStyled-sc-9xia41-0 TitleStyled-sc-t58pjn-0 dZypbt dXQqcD"
          color="#FFF"
          font-family="AdobeCleanBold"
          font-size="24"
          letter-spacing="0.9"
          width="100%"
        >
          Success
        </h3>
        <p
          class="TextStyled-sc-9xia41-0 jJKZfr"
          color="#EAEAEB"
          font-size="13"
          font-weight="300"
          letter-spacing="0.9"
          width="100%"
        >
          Your changes have been successfully saved.
        </p>
      </div>
    </div>
    <button
      class="ButtonStyled-sc-1ys9v52-0 bRMDWm"
      data-testid="button"
    >
      <span
        class="ButtonStyled__ButtonIconStyled-sc-1ys9v52-1 DjHyU"
      >
        <div
          class="ViewStyled-sc-rqe5sa-0 efewJe"
          display="inline-block"
          height="15"
          width="15"
        >
          <span
            style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
          >
            <img
              data-nimg="fill"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
      </span>
    </button>
  </div>
</DocumentFragment>
`;
